src/contracts/BCoWHelper.sol

```diff
@@ -117,9 +97,97 @@ contract BCoWHelper is ICOWAMMPoolHelper, BMath {
     if (tokens_.length != 2) {
       revert PoolDoesNotExist();
     }
-    // reverts in case pool is not supported (non-equal weights)
-    if (IBCoWPool(pool).getNormalizedWeight(tokens_[0]) != IBCoWPool(pool).getNormalizedWeight(tokens_[1])) {
-      revert PoolDoesNotExist();
+  }
+
+  /// @notice Returns the order that is suggested to be executed to CoW Protocol
+  /// for a specific pool given the current chain state and final token prices
+  /// @param pool The pool whose funds should be used in the order
+  /// @param prices The prices for each token of the pool
+  /// @return order_ The suggested CoW Protocol order
+  function _rawOrder(address pool, uint256[] calldata prices) internal view returns (GPv2Order.Data memory order_) {
+    address[] memory tokenPair = tokens(pool);
+    Reserves memory reservesOut = _reserves(IBCoWPool(pool), IERC20(tokenPair[0]));
+    Reserves memory reservesIn = _reserves(IBCoWPool(pool), IERC20(tokenPair[1]));
+
+    // The out amount is computed according to the following formula:
+    // aO = amountOut
+    // bI = reservesIn.balance                   bO * wI - p * bI * wO
+    // bO = reservesOut.balance            aO =  ---------------------
+    // wI = reservesIn.denormWeight                     wI + wO
+    // wO = reservesOut.denormWeight
+    // p  = priceNumerator / priceDenominator
+    //
+    // Note that in the code we use normalized weights instead of computing the
+    // full expression from raw weights. This is equivalent to assuming that
+    // wI + wO = 1.
```

Comment:
```suggestion
    // full expression from raw weights. Since BCoW pools support only two
    // tokens, this is equivalent to assuming that wI + wO = 1.
```